security/libressl/Portfile

```diff
@@ -23,9 +23,9 @@ homepage            https://www.libressl.org
 conflicts           openssl libressl-devel
 
 master_sites        openbsd:LibreSSL
-checksums           rmd160  8d1ab600e5d3547cef3b04a5cd49a9e99d1ac99e \
-                    sha256  4be80fff073746cf50b4a8e5babe2795acae98c6b132a9e02519b445dfbfd033 \
-                    size    4239482
+checksums           rmd160  f8387bd66d72776c58aaaf7a040c0f325bd75d4f \
+                    sha256  b06aa538fefc9c6b33c4db4931a09a5f52d9d2357219afcbff7d93fe12ebf6f7 \
+                    size    4336143
 
 patchfiles          patch-openssldir-cert.pem.diff
```

Comment:
> ```
> Hunk #1 succeeded at 2 with fuzz 1 (offset -4 lines).
> ```

Please de-fuzz the patchfile. The `patch` program on Ventura is pickier than on earlier macOS versions and often will not apply fuzzy patches.

Reply:
Ah, good to know. I don't have Ventura installed anywhere at present to test this. Looking over the patch (which is also in libressl-devel) and the motivating ticket which created it: https://trac.macports.org/ticket/51807 I am wondering if there might be a better way to address this without the patch at all? Admittedly, I am just sitting down to look at this after having driven 1000+ miles in the last couple of days so I am not in the freshest frame of mind at the moment. Maybe @neverpanic has some more expert eyeballs to assist on this one?

Reply:
(So tired, I wasn't even identifying that I am actually already running Ventura; for some reason my brain was saying "Monterey" I guess?)

Attempting to apply the patch manually, I don't see the error encountered, but I am able to simulate it sort of if I run patch with -F 0.

e.g.

```
% patch -F 0 opensslconf.h patch-openssldir-cert.pem.diff
patching file opensslconf.h
1 out of 1 hunks failed--saving rejects to opensslconf.h.rej
```

I am probably still a bit too tired and unfocused to rectify this at the moment. I hopefully will have some more time and energy to do so tomorrow afternoon.

Reply:
I *think* this patch may be de-fuzzed as follows?

```
--- include/openssl/opensslconf.h.orig
+++ include/openssl/opensslconf.h
@@ -2,7 +2,7 @@
 /* crypto/opensslconf.h.in */
 
 #if defined(HEADER_CRYPTLIB_H) && !defined(OPENSSLDIR)
-#define OPENSSLDIR "/etc/ssl"
+#define OPENSSLDIR "@OPENSSLDIR@"
 #endif
 
 #undef OPENSSL_UNISTD
```

At least, testing locally.

e.g.

```
% patch -F 0 opensslconf.h patch-openssldir-cert.pem.diff    
patching file opensslconf.h
1 out of 1 hunks failed--saving rejects to opensslconf.h.rej
```

Whereas the revised diff using the changes above:

```
% patch -F 0 opensslconf.h patch-openssldir-cert.pem.diff.new 
patching file opensslconf.h
```

Seems to apply cleanly?

Reply:
OK, my last push includes the de-fuzzed diff. Hopefully that rectifies the concerns addressed here?